scene/resources/visual_shader.cpp

```diff
@@ -951,7 +952,7 @@ bool VisualShader::can_connect_nodes(Type p_type, int p_from_node, int p_from_po
 }
 
 bool VisualShader::is_port_types_compatible(int p_a, int p_b) const {
-	return MAX(0, p_a - 5) == (MAX(0, p_b - 5));
+	return MAX(0, p_a - 6) == (MAX(0, p_b - 6));
```

Comment:
We should consider replacing the 6 with ``PORT_TYPE_BOOLEAN``. If we end up adding more types it would be nice not to have to update this each time (i.e. doubles, uvec*, ivec* etc.)